pkg/vpcmodel/detailedConn.go

```diff
@@ -0,0 +1,113 @@
+/*
+Copyright 2023- IBM Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package vpcmodel
+
+import (
+	"github.com/np-guard/models/pkg/connection"
+	"github.com/np-guard/models/pkg/netp"
+)
+
+// todo: remove stateful from connection.Set
+
+// detailedConn captures the connection with connection's responsive details, as described below.
+// It is created from src to dest allowed connection (TCP and non-TCP) and response dest to src allowed connection
+// (TCP and non-TCP); further entities of the connection may be created from operations as union e.g. for abstraction
+// note: tcpRspDisable is not independent and is calculated based on the other properties;
+// it is kept since it is widely used - to determine if the connection is responsive
+type detailedConn struct {
+	tcpRspEnable  *connection.Set // responsive TCP connection between <src, dst>
+	nonTCP        *connection.Set // non TCP connection (for which responsiveness is non-relevant)
+	allConn       *connection.Set // entire connection
+	tcpRspDisable *connection.Set // non-responsive TCP connection between <src, dst>; complementary of tcpRspEnable
+	// connection is defined to be responsive if nonTCP is empty
+}
+
+// operation on detailedConn
+// The operations are performed on the disjoint tcpRspEnable and nonTCP and on allConn which contains them;
+// tcpRspDisable - the tcp complementary of tcpRspEnable w.r.t. allConn -
+// is computed as allConn minus (tcpRspEnable union nonTCP)
+
+func newDetailConn(tspRspConn, otherConn, allConn *connection.Set) *detailedConn {
+	return &detailedConn{
+		tcpRspEnable:  tspRspConn,
+		tcpRspDisable: (allConn.Subtract(otherConn)).Subtract(tspRspConn),
+		nonTCP:        otherConn,
+		allConn:       allConn,
+	}
+}
+
+func emptyDetailConn() *detailedConn {
+	return newDetailConn(NoConns(), NoConns(), NoConns())
```

Comment:
```suggestion
	return newDetailedConn(NoConns(), NoConns(), NoConns())
```

and in the rest of the functions here

Reply:
done